tools/rpkg/R/backend-dbplyr__duckdb_connection.R

```diff
@@ -0,0 +1,383 @@
+        if (fiscal_start != 1) {
+          stop("`fiscal_start` is not yet supported in DuckDB translation. Must be 1.", call. = FALSE)
+        }
+        if (is.logical(type)) {
+          type <- if (type) {
+            "year.quarter"
+          } else {
+            "quarter"
+          }
+        }
+        if (with_year == TRUE) {
+          type <- "year.quarter"
+        }
+        switch(type,
+          quarter = {
+            sql_expr(EXTRACT(QUARTER %FROM% !!x))
+          },
+          year.quarter = {
+            sql_expr((EXTRACT(YEAR %FROM% !!x) || "." || EXTRACT(QUARTER %FROM% !!x)))
+          },
+          date_first = {
+            sql_expr((CAST(DATE_TRUNC("QUARTER", !!x) %AS% DATE)))
+          },
+          date_last = {
+            build_sql("(CAST((DATE_TRUNC('QUARTER',", x, ") + INTERVAL '1 QUARTER' - INTERVAL '1 DAY') AS DATE))")
+          },
+          stop(paste("Unsupported type", type), call. = FALSE)
+        )
+      },
+      qday = function(x) {
+        build_sql("DATE_DIFF('DAYS',DATE_TRUNC('QUARTER',CAST((", x, ") AS DATE)),(CAST((", x, ") AS DATE) + INTERVAL '1 DAY'))")
+      },
+      wday = function(x, label = FALSE, abbr = TRUE, week_start = NULL) {
+        if (!label) {
+          week_start <- if (!is.null(week_start)) week_start else getOption("lubridate.week.start", 7)
+          offset <- as.integer(7 - week_start)
+          sql_expr(EXTRACT("dow" %FROM% CAST((!!x) %AS% DATE) + !!offset) + 1L)
+        } else if (label && !abbr) {
+          sql_expr(STRFTIME(!!x, "%A"))
+        } else if (label && abbr) {
+          sql_expr(STRFTIME(!!x, "%a"))
+        } else {
+          stop("Unrecognized arguments to `wday`", call. = FALSE)
+        }
+      },
+      yday = function(x) sql_expr(EXTRACT(DOY %FROM% !!x)),
+
+      # These work fine internally, but getting INTERVAL-type data out of DuckDB,
+      # cast to CHARs is required until there is a fix for the issue #2900
+      # (https://github.com/duckdb/duckdb/issues/2900)
+      seconds = function(x) {
+        sql_expr(TO_SECONDS(CAST((!!x) %AS% BIGINT)))
+      },
+      minutes = function(x) {
+        sql_expr(TO_MINUTES(CAST((!!x) %AS% BIGINT)))
+      },
+      hours = function(x) {
+        sql_expr(TO_HOURS(CAST((!!x) %AS% BIGINT)))
+      },
+      days = function(x) {
+        sql_expr(TO_DAYS(CAST((!!x) %AS% INTEGER)))
+      },
+      weeks = function(x) {
+        sql_expr(TO_DAYS(7L * CAST((!!x) %AS% INTEGER)))
+      },
+      months = function(x) {
+        sql_expr(TO_MONTHS(CAST((!!x) %AS% INTEGER)))
+      },
+      years = function(x) {
+        sql_expr(TO_YEARS(CAST((!!x) %AS% INTEGER)))
+      },
+
+      # Type may change from date to timestamp here now that is not consistent with the R-function
+      # Week_start not implemented and default is different from R-function (1 [Monday] instead of 7 [Sunday])
```

Comment:
Tried to do that, but it didn't work as easily as with weekday, because the logic in determining how the weeks fit to the year changes. I'll take a one more look.

Reply:
Found an algorithm used in lubridate (tidyverse/lubridate#509) and now the `week_start` argument is implemented.